src/core/olimageryprovider.js

```diff
@@ -208,12 +208,9 @@ olcs.core.OLImageryProvider.prototype.requestImage = function(x, y, level) {
   var tileUrlFunction = this.source_.getTileUrlFunction();
   if (!goog.isNull(tileUrlFunction) && !goog.isNull(this.projection_)) {
     // perform mapping of Cesium tile coordinates to ol3 tile coordinates
-    var z_ = (this.tilingScheme_ instanceof Cesium.GeographicTilingScheme) ?
-             (level + 1) : level;
-    var y_ = (this.source_.getTileGrid() instanceof ol.tilegrid.XYZ) ?
-             y : (y - (1 << level));
-    y_ = -y_ - 1; // opposite indexing
-    var url = tileUrlFunction([z_, x, y_], 1, this.projection_);
+    var transform = this.source_.getTileGrid().createTileCoordTransform();
+    var tileCoord = transform([level, x, y]);
```

Comment:
You have to do the transform on the original tileCoord, before changing the level. This might also explain your issue with TileWMS.